File: Bon_plans_desktop_app/src/tn/esprit/bonplans/entity/Plan.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package tn.esprit.bonplans.entity;

import java.util.Date;

/**
 *
 * @author SadfiAmine
 */

public class Plan {
    private int idPlan;
    private String titre ;
    private String description;
    private  String urlPhoto;
    private  Double prixInitial;
    private Double prixPromo;
    private int nbPlaceTotal;
    private Date dateDebut;
    private Date dateFin;
    private int nbPlaceDispo;
    private int statut;
    private int like;
    private int dislike;
    
    //foreign keys
    private int idAnnonceur;
    private int idCategorie;

    public Plan() {
    }

    public Plan(int idPlan, String titre, String description, String urlPhoto, Double prixInitial, Double prixPromo, int nbPlaceTotal, Date dateDebut, Date dateFin, int nbPlaceDispo, int statut, int idAnnonceur, int idCategorie) {
        this.idPlan = idPlan;
        this.titre = titre;
        this.description = description;
        this.urlPhoto = urlPhoto;
        this.prixInitial = prixInitial;
        this.prixPromo = prixPromo;
        this.nbPlaceTotal = nbPlaceTotal;
        this.dateDebut = dateDebut;
        this.dateFin = dateFin;
        this.nbPlaceDispo = nbPlaceDispo;
        this.statut = statut;
        this.idAnnonceur = idAnnonceur;
        this.idCategorie = idCategorie;
    }

    public Plan(String titre, String description, String urlPhoto, Double prixInitial, Double prixPromo, int nbPlaceTotal, Date dateDebut, Date dateFin, int nbPlaceDispo, int statut, int idAnnonceur, int idCategorie) {
        this.titre = titre;
        this.description = description;
        this.urlPhoto = urlPhoto;
        this.prixInitial = prixInitial;
        this.prixPromo = prixPromo;
        this.nbPlaceTotal = nbPlaceTotal;
        this.dateDebut = dateDebut;
        this.dateFin = dateFin;
        this.nbPlaceDispo = nbPlaceDispo;
        this.statut = statut;
        this.idAnnonceur = idAnnonceur;
        this.idCategorie = idCategorie;
    }

    public int getIdPlan() {
        return idPlan;
    }

    public void setIdPlan(int idPlan) {
        this.idPlan = idPlan;
    }

    public String getTitre() {
        return titre;
    }

    public void setTitre(String titre) {
        this.titre = titre;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getUrlPhoto() {
        return urlPhoto;
    }

    public void setUrlPhoto(String urlPhoto) {
        this.urlPhoto = urlPhoto;
    }

    public Double getPrixInitial() {
        return prixInitial;
    }

    public void setPrixInitial(Double prixInitial) {
        this.prixInitial = prixInitial;
    }

    public Double getPrixPromo() {
        return prixPromo;
    }

    public void setPrixPromo(Double prixPromo) {
        this.prixPromo = prixPromo;
    }

    public int getNbPlaceTotal() {
        return nbPlaceTotal;
    }

    public void setNbPlaceTotal(int nbPlaceTotal) {
        this.nbPlaceTotal = nbPlaceTotal;
    }

    public Date getDateDebut() {
        return dateDebut;
    }

    public void setDateDebut(Date dateDebut) {
        this.dateDebut = dateDebut;
    }

    public Date getDateFin() {
        return dateFin;
    }

    public void setDateFin(Date dateFin) {
        this.dateFin = dateFin;
    }

    public int getNbPlaceDispo() {
        return nbPlaceDispo;
    }

    public void setNbPlaceDispo(int nbPlaceDispo) {
        this.nbPlaceDispo = nbPlaceDispo;
    }

    public int getStatut() {
        return statut;
    }

    public void setStatut(int statut) {
        
        this.statut = statut;
    }

    public int getIdAnnonceur() {
        return idAnnonceur;
    }

    public void setIdAnnonceur(int idAnnonceur) {
        this.idAnnonceur = idAnnonceur;
    }

    public int getIdCategorie() {
        return idCategorie;
    }

    public void setIdCategorie(int idCategorie) {
        this.idCategorie = idCategorie;
    }

    public int getLike() {
        return like;
    }

    public void setLike(int like) {
        this.like = like;
    }

    public int getDislike() {
        return dislike;
    }

    public void setDislike(int dislike) {
        this.dislike = dislike;
    }

    @Override
    public String toString() {
        return titre;
    }
    
    @Override
    public int hashCode() {
        int hash = 7;
        hash = 59 * hash + this.idPlan;
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Plan other = (Plan) obj;
        if (this.idPlan != other.idPlan) {
            return false;
        }
        return true;
    }

    
    
}

File: www/Bon_plans_web_app/Entity/Plan.php
<?php
/**
 * Created by PhpStorm.
 * User: SadfiAmine
 * Date: 15/04/2018
 * Time: 13:32
 */

class Plan
{
    private  $idPlan;
    private  $titre ;
    private  $description;
    private  $urlPhoto;
    private  $prixInitial;
    private  $prixPromo;
    private  $nbPlaceTotal;
    private  $dateDebut;
    private  $dateFin;
    private  $nbPlaceDispo;
    private  $statut;
    private  $nbJaime ;
    private  $nbJaimePas;
    private  $note ;
        //
    private  $idClient;
    private  $idAnnonceur;
    private  $idCategorie;

    public function __construct()
    {
    }
    public function __construct1($idPlan,$titre,$description,$urlPhoto,$prixInitial,$prixPromo,$nbPlaceTotal,$dateDebut,$dateFin,$nbPlaceDispo,$statut,$nbJaime,$nbJaimePas,$note,$idClient,$idAnnonceur,$idCategorie)
    {
        $this->$idPlan=$idPlan;
        $this->$titre=$titre;
        $this->$description=$description;
        $this->$urlPhoto=$urlPhoto;
        $this->$prixInitial=$prixInitial;
        $this->$prixPromo=$prixPromo;
        $this->$nbPlaceTotal=$nbPlaceTotal;
        $this->$dateDebut=$dateDebut;
        $this->$dateFin=$dateFin;
        $this->$nbPlaceDispo=$nbPlaceDispo;
        $this->$statut=$statut;
        $this->$nbJaime=$nbJaime;
        $this->$nbJaimePas=$nbJaimePas;
        $this->$note=$note;
        $this->$idClient=$idClient;
        $this->$idAnnonceur=$idAnnonceur;
        $this->$idCategorie=$idCategorie;
    }
    public function __construct2($titre,$description,$urlPhoto,$prixInitial,$prixPromo,$nbPlaceTotal,$dateDebut,$dateFin,$nbPlaceDispo,$statut,$nbJaime,$nbJaimePas,$note,$idClient,$idAnnonceur,$idCategorie)
    {
        $this->$titre=$titre;
        $this->$description=$description;
        $this->$urlPhoto=$urlPhoto;
        $this->$prixInitial=$prixInitial;
        $this->$prixPromo=$prixPromo;
        $this->$nbPlaceTotal=$nbPlaceTotal;
        $this->$dateDebut=$dateDebut;
        $this->$dateFin=$dateFin;
        $this->$nbPlaceDispo=$nbPlaceDispo;
        $this->$statut=$statut;
        $this->$nbJaime=$nbJaime;
        $this->$nbJaimePas=$nbJaimePas;
        $this->$note=$note;
        $this->$idClient=$idClient;
        $this->$idAnnonceur=$idAnnonceur;
        $this->$idCategorie=$idCategorie;
    }

    /**
     * @return mixed
     */
    public function getIdPlan()
    {
        return $this->idPlan;
    }

    /**
     * @param mixed $idPlan
     */
    public function setIdPlan($idPlan)
    {
        $this->idPlan = $idPlan;
    }

    /**
     * @return mixed
     */
    public function getDescription()
    {
        return $this->description;
    }

    /**
     * @param mixed $description
     */
    public function setDescription($description)
    {
        $this->description = $description;
    }

    /**
     * @param mixed $idClient
     */
    public function setIdClient($idClient)
    {
        $this->idClient = $idClient;
    }

    /**
     * @return mixed
     */
    public function getIdClient()
    {
        return $this->idClient;
    }

    /**
     * @param mixed $nbJaimePas
     */
    public function setNbJaimePas($nbJaimePas)
    {
        $this->nbJaimePas = $nbJaimePas;
    }

    /**
     * @return mixed
     */
    public function getNbJaimePas()
    {
        return $this->nbJaimePas;
    }

    /**
     * @return mixed
     */
    public function getTitre()
    {
        return $this->titre;
    }

    /**
     * @param mixed $titre
     */
    public function setTitre($titre)
    {
        $this->titre = $titre;
    }

    /**
     * @param mixed $idCategorie
     */
    public function setIdCategorie($idCategorie)
    {
        $this->idCategorie = $idCategorie;
    }

    /**
     * @return mixed
     */
    public function getIdCategorie()
    {
        return $this->idCategorie;
    }

    /**
     * @return mixed
     */
    public function getNbJaime()
    {
        return $this->nbJaime;
    }

    /**
     * @param mixed $nbJaime
     */
    public function setNbJaime($nbJaime)
    {
        $this->nbJaime = $nbJaime;
    }

    /**
     * @return mixed
     */
    public function getDateDebut()
    {
        return $this->dateDebut;
    }

    /**
     * @param mixed $dateDebut
     */
    public function setDateDebut($dateDebut)
    {
        $this->dateDebut = $dateDebut;
    }

    /**
     * @return mixed
     */
    public function getDateFin()
    {
        return $this->dateFin;
    }

    /**
     * @param mixed $dateFin
     */
    public function setDateFin($dateFin)
    {
        $this->dateFin = $dateFin;
    }

    /**
     * @return mixed
     */
    public function getIdAnnonceur()
    {
        return $this->idAnnonceur;
    }

    /**
     * @param mixed $idAnnonceur
     */
    public function setIdAnnonceur($idAnnonceur)
    {
        $this->idAnnonceur = $idAnnonceur;
    }

    /**
     * @return mixed
     */
    public function getNbPlaceDispo()
    {
        return $this->nbPlaceDispo;
    }

    /**
     * @param mixed $nbPlaceDispo
     */
    public function setNbPlaceDispo($nbPlaceDispo)
    {
        $this->nbPlaceDispo = $nbPlaceDispo;
    }

    /**
     * @return mixed
     */
    public function getNbPlaceTotal()
    {
        return $this->nbPlaceTotal;
    }

    /**
     * @param mixed $nbPlaceTotal
     */
    public function setNbPlaceTotal($nbPlaceTotal)
    {
        $this->nbPlaceTotal = $nbPlaceTotal;
    }

    /**
     * @return mixed
     */
    public function getNote()
    {
        return $this->note;
    }

    /**
     * @param mixed $note
     */
    public function setNote($note)
    {
        $this->note = $note;
    }

    /**
     * @return mixed
     */
    public function getPrixInitial()
    {
        return $this->prixInitial;
    }

    /**
     * @param mixed $prixInitial
     */
    public function setPrixInitial($prixInitial)
    {
        $this->prixInitial = $prixInitial;
    }

    /**
     * @return mixed
     */
    public function getPrixPromo()
    {
        return $this->prixPromo;
    }

    /**
     * @param mixed $prixPromo
     */
    public function setPrixPromo($prixPromo)
    {
        $this->prixPromo = $prixPromo;
    }

    /**
     * @return mixed
     */
    public function getStatut()
    {
        return $this->statut;
    }

    /**
     * @param mixed $statut
     */
    public function setStatut($statut)
    {
        $this->statut = $statut;
    }

    /**
     * @return mixed
     */
    public function getUrlPhoto()
    {
        return $this->urlPhoto;
    }

    /**
     * @param mixed $urlPhoto
     */
    public function setUrlPhoto($urlPhoto)
    {
        $this->urlPhoto = $urlPhoto;
    }


}